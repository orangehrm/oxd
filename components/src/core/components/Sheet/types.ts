export const TYPE_GRAY_DARKEN_2 = 'gray-darken-2';
export const TYPE_GRAY_DARKEN_1 = 'gray-darken-1';
export const TYPE_GRAY = 'gray';
export const TYPE_GRAY_LIGHTEN_1 = 'gray-lighten-1';
export const TYPE_GRAY_LIGHTEN_2 = 'gray-lighten-2';
export const TYPE_LIGHT_GRAY = 'light-gray';
export const TYPE_PASTEL_WHITE = 'pastel-white';
export const TYPE_WHITE_SHADOW = 'white-shadow';
export const TYPE_WHITE = 'white';

export const TYPES = [
  TYPE_GRAY_DARKEN_2,
  TYPE_GRAY_DARKEN_1,
  TYPE_GRAY,
  TYPE_GRAY_LIGHTEN_1,
  TYPE_GRAY_LIGHTEN_2,
  TYPE_LIGHT_GRAY,
  TYPE_PASTEL_WHITE,
  TYPE_WHITE_SHADOW,
  TYPE_WHITE,
];

export type SheetType =
  | typeof TYPE_GRAY_DARKEN_2
  | typeof TYPE_GRAY_DARKEN_1
  | typeof TYPE_GRAY
  | typeof TYPE_GRAY_LIGHTEN_1
  | typeof TYPE_GRAY_LIGHTEN_2
  | typeof TYPE_LIGHT_GRAY
  | typeof TYPE_PASTEL_WHITE
  | typeof TYPE_WHITE_SHADOW
  | typeof TYPE_WHITE;
