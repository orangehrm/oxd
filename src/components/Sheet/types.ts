/*
 * This file is part of OrangeHRM Inc
 *
 * Copyright (C) 2020 onwards OrangeHRM Inc
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see  http://www.gnu.org/licenses
 */

export const TYPE_GRAY_DARKEN_2 = 'gray-darken-2';
export const TYPE_GRAY_DARKEN_1 = 'gray-darken-1';
export const TYPE_GRAY = 'gray';
export const TYPE_GRAY_LIGHTEN_1 = 'gray-lighten-1';
export const TYPE_GRAY_LIGHTEN_2 = 'gray-lighten-2';
export const TYPE_LIGHT_GRAY = 'light-gray';
export const TYPE_PASTEL_WHITE = 'pastel-white';
export const TYPE_WHITE_SHADOW = 'white-shadow';
export const TYPE_WHITE = 'white';

export const TYPES = [
  TYPE_GRAY_DARKEN_2,
  TYPE_GRAY_DARKEN_1,
  TYPE_GRAY,
  TYPE_GRAY_LIGHTEN_1,
  TYPE_GRAY_LIGHTEN_2,
  TYPE_LIGHT_GRAY,
  TYPE_PASTEL_WHITE,
  TYPE_WHITE_SHADOW,
  TYPE_WHITE,
];

export type SheetType =
  | typeof TYPE_GRAY_DARKEN_2
  | typeof TYPE_GRAY_DARKEN_1
  | typeof TYPE_GRAY
  | typeof TYPE_GRAY_LIGHTEN_1
  | typeof TYPE_GRAY_LIGHTEN_2
  | typeof TYPE_LIGHT_GRAY
  | typeof TYPE_PASTEL_WHITE
  | typeof TYPE_WHITE_SHADOW
  | typeof TYPE_WHITE;
