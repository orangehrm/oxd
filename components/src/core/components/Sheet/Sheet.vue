<template>
  <div :class="classes">
    <slot></slot>
  </div>
</template>

<script lang="ts">
import {defineComponent} from 'vue';
import {SheetType, TYPES, TYPE_WHITE} from './types';

export default defineComponent({
  name: 'oxd-sheet',

  props: {
    type: {
      type: String,
      default: TYPE_WHITE,
      validator: function(value: SheetType) {
        return TYPES.indexOf(value) !== -1;
      },
    },
    rounded: {
      type: Boolean,
      default: true,
    },
    gutters: {
      type: Boolean,
      default: true,
    },
  },

  computed: {
    classes(): object {
      return {
        'oxd-sheet': true,
        'oxd-sheet--rounded': this.rounded,
        'oxd-sheet--gutters': this.gutters,
        [`oxd-sheet--${this.type}`]: true,
      };
    },
  },
});
</script>

<style src="./sheet.scss" lang="scss" scoped></style>
