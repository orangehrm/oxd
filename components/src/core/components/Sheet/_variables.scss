/* OXD - Interface Colors */
$oxd-sheet-gray-darken-2-color: $oxd-interface-gray-darken-2-color !default;
$oxd-sheet-gray-darken-1-color: $oxd-interface-gray-darken-1-color !default;
$oxd-sheet-gray-color: $oxd-interface-gray-color !default;
$oxd-sheet-gray-lighten-1-color: $oxd-interface-gray-lighten-1-color !default;
$oxd-sheet-gray-lighten-2-color: $oxd-interface-gray-lighten-2-color !default;

/* OXD - Background & Overly Colors */
$oxd-sheet-light-gray-color: $oxd-background-light-gray-color !default;
$oxd-sheet-pastel-white-color: $oxd-background-pastel-white-color !default;
$oxd-sheet-white-shadow-color: $oxd-background-white-shadow-color !default;

$oxd-sheet-white-color: $oxd-white-color !default;

$oxd-sheet-border-radius: $oxd-border-radius !default;
$oxd-sheet-padding-vertical: 1.2rem !default;
$oxd-sheet-padding-horizontal: 1.2rem !default;
