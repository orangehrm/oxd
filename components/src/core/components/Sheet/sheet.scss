@import '../../../styles';
@import 'variables';

.oxd-sheet {
  background-color: $oxd-sheet-white-color;

  &--gray-darken-2 {
    background-color: $oxd-sheet-gray-darken-2-color;
  }
  &--gray-darken-1 {
    background-color: $oxd-sheet-gray-darken-1-color;
  }
  &--gray {
    background-color: $oxd-sheet-gray-color;
  }
  &--gray-lighten-1 {
    background-color: $oxd-sheet-gray-lighten-1-color;
  }
  &--gray-lighten-2 {
    background-color: $oxd-sheet-gray-lighten-2-color;
  }
  &--light-gray {
    background-color: $oxd-sheet-light-gray-color;
  }
  &--pastel-white {
    background-color: $oxd-sheet-pastel-white-color;
  }
  &--white-shadow {
    background-color: $oxd-sheet-white-shadow-color;
  }

  &--rounded {
    border-radius: $oxd-sheet-border-radius;
  }

  &--gutters {
    padding-top: $oxd-sheet-padding-vertical;
    padding-bottom: $oxd-sheet-padding-vertical;
    padding-left: $oxd-sheet-padding-horizontal;
    padding-right: $oxd-sheet-padding-horizontal;
  }
}
