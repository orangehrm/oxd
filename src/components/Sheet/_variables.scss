// This file is part of OrangeHRM Inc
//
// Copyright (C) 2020 onwards OrangeHRM Inc
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see  http://www.gnu.org/licenses

// OXD - Interface Colors
$oxd-sheet-gray-darken-2-color: $oxd-interface-gray-darken-2-color !default;
$oxd-sheet-gray-darken-1-color: $oxd-interface-gray-darken-1-color !default;
$oxd-sheet-gray-color: $oxd-interface-gray-color !default;
$oxd-sheet-gray-lighten-1-color: $oxd-interface-gray-lighten-1-color !default;
$oxd-sheet-gray-lighten-2-color: $oxd-interface-gray-lighten-2-color !default;

// OXD - Background & Overly Colors
$oxd-sheet-light-gray-color: $oxd-background-light-gray-color !default;
$oxd-sheet-pastel-white-color: $oxd-background-pastel-white-color !default;
$oxd-sheet-white-shadow-color: $oxd-background-white-shadow-color !default;

$oxd-sheet-white-color: $oxd-white-color !default;

$oxd-sheet-border-radius: $oxd-border-radius !default;
$oxd-sheet-padding-vertical: 25px !default;
$oxd-sheet-padding-horizontal: 25px !default;
