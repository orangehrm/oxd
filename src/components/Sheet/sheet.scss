// This file is part of OrangeHRM Inc
//
// Copyright (C) 2020 onwards OrangeHRM Inc
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see  http://www.gnu.org/licenses

@import 'variables';

.oxd-sheet {
  background-color: $oxd-sheet-white-color;

  &--gray-darken-2 {
    background-color: $oxd-sheet-gray-darken-2-color;
  }
  &--gray-darken-1 {
    background-color: $oxd-sheet-gray-darken-1-color;
  }
  &--gray {
    background-color: $oxd-sheet-gray-color;
  }
  &--gray-lighten-1 {
    background-color: $oxd-sheet-gray-lighten-1-color;
  }
  &--gray-lighten-2 {
    background-color: $oxd-sheet-gray-lighten-2-color;
  }
  &--light-gray {
    background-color: $oxd-sheet-light-gray-color;
  }
  &--pastel-white {
    background-color: $oxd-sheet-pastel-white-color;
  }
  &--white-shadow {
    background-color: $oxd-sheet-white-shadow-color;
  }

  &--rounded {
    border-radius: $oxd-sheet-border-radius;
  }

  &--gutters {
    padding-top: $oxd-sheet-padding-vertical;
    padding-bottom: $oxd-sheet-padding-vertical;
    padding-left: $oxd-sheet-padding-horizontal;
    padding-right: $oxd-sheet-padding-horizontal;
  }
}
