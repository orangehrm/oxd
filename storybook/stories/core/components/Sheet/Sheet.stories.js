import Sheet from '@orangehrm/oxd/core/components/Sheet/Sheet';
import {TYPES} from '@orangehrm/oxd/core/components/Sheet/types';
import SheetWithGuttersStory from './SheetWithGutters.story.vue';
import SheetWithBackgroundStory from './SheetWithBackground.story.vue';

export default {
  title: 'Example/Sheet',
  component: Sheet,
  argTypes: {
    type: {
      control: {type: 'select', options: TYPES},
    },
    rounded: {
      control: {type: 'boolean'},
    },
    gutters: {
      control: {type: 'boolean'},
    },
  },
};

const Template = args => ({
  setup() {
    return {args};
  },
  components: {'oxd-sheet': Sheet},
  template: '<oxd-sheet v-bind="args" />',
});

export const Default = Template.bind({});
Default.args = {
  type: 'gray',
  rounded: true,
  style: {
    width: '400px',
    height: '300px',
  },
};

export const SheetWithGutters = () => SheetWithGuttersStory;
SheetWithGutters.args = {
  type: 'gray-lighten-2',
  rounded: true,
  gutters: true,
};

export const SheetWithBackground = () => SheetWithBackgroundStory;
SheetWithBackground.args = {
  rounded: true,
};
