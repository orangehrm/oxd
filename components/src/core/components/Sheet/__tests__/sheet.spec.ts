import {mount} from '@vue/test-utils';
import Sheet from '@orangehrm/oxd/core/components/Sheet/Sheet.vue';

describe('Sheet.vue', () => {
  it('should renders OXD Sheet', () => {
    const wrapper = mount(Sheet, {
      props: {},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Sheet, not rounded', () => {
    const wrapper = mount(Sheet, {
      props: {rounded: false},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Sheet, without gutters', () => {
    const wrapper = mount(Sheet, {
      props: {gutters: false},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Sheet, type white', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'white'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Sheet, type white-shadow', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'white-shadow'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Sheet, type pastel-white', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'pastel-white'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Sheet, type light-gray', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'light-gray'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Sheet, type gray-lighten-2', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'gray-lighten-2'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Sheet, type gray-lighten-1', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'gray-lighten-1'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Sheet, type gray', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'gray'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Sheet, type gray-darken-1', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'gray-darken-1'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Sheet, type gray-darken-2', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'gray-darken-2'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
