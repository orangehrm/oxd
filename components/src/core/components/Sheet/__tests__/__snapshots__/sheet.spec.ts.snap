// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sheet.vue should renders OXD Sheet 1`] = `<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--white"></div>`;

exports[`Sheet.vue should renders OXD Sheet, not rounded 1`] = `<div class="oxd-sheet oxd-sheet--gutters oxd-sheet--white"></div>`;

exports[`Sheet.vue should renders OXD Sheet, type gray 1`] = `<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--gray"></div>`;

exports[`Sheet.vue should renders OXD Sheet, type gray-darken-1 1`] = `<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--gray-darken-1"></div>`;

exports[`Sheet.vue should renders OXD Sheet, type gray-darken-2 1`] = `<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--gray-darken-2"></div>`;

exports[`Sheet.vue should renders OXD Sheet, type gray-lighten-1 1`] = `<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--gray-lighten-1"></div>`;

exports[`Sheet.vue should renders OXD Sheet, type gray-lighten-2 1`] = `<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--gray-lighten-2"></div>`;

exports[`Sheet.vue should renders OXD Sheet, type light-gray 1`] = `<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--light-gray"></div>`;

exports[`Sheet.vue should renders OXD Sheet, type pastel-white 1`] = `<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--pastel-white"></div>`;

exports[`Sheet.vue should renders OXD Sheet, type white 1`] = `<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--white"></div>`;

exports[`Sheet.vue should renders OXD Sheet, type white-shadow 1`] = `<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--white-shadow"></div>`;

exports[`Sheet.vue should renders OXD Sheet, without gutters 1`] = `<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--white"></div>`;
